mitodl/mit-open
# Additional offeror details

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/4273

### Description (What does it do?)
The `offered_by` element of serialized learning resource data should be the same as before:
http://localhost:8063/api/v1/learning_resources/?offered_by=see
```json
            "offered_by": {
                "code": "see",
                "name": "Sloan Executive Education",
                "channel_url": "http://mbod.odl.local:8063/c/offeror/see/"
            },
```

But there should be more attributes included in the `/api/v1/offerors` API response:
http://localhost:8063/api/v1/offerors
```json
{
            "code": "see",
            "channel_url": "http://mbod.odl.local:8063/c/offeror/see/",
            "name": "Sloan Executive Education",
            "professional": true,
            "offerings": [
                "Courses",
                "Programs",
                "Enterprise or Organizational accounts",
                "Custom Engagements"
            ],
            "audience": [
                "Global Business Professionals (all levels)",
                "Entrepreneurs and intrapreneurs",
                "Managers and executives",
                "Current and future business leaders"
            ],
            "formats": [
                "Online",
                "In-Person",
                "Blended"
            ],
            "fee": [
                "Paid",
                "Free Webinars and Blogs"
            ],
            "certifications": [
                "Executive Certificates",
                "Professional Certificates"
            ],
            "content_types": [
                "Professional"
            ],
            "more_information": [
                "https://executive.mit.edu/"
            ]
        },
```

and these attributes should also be included in the channels API response for an offeror channel:
http://localhost:8063/api/v0/channels/?channel_type=offeror
```json
{
            "id": 375,
            "avatar": null,
            "avatar_small": null,
            "avatar_medium": null,
            "banner": null,
            "is_moderator": false,
            "lists": [],
            "channel_url": "http://mbod.odl.local:8063/c/offeror/see/",
            "featured_list": {
                "title": "Sloan Executive Education Featured Resources",
                "url": null,
                "id": 3313
            },
            "subfields": [],
            "channel_type": "offeror",
            "offeror_detail": {
                "offeror": {
                    "code": "see",
                    "channel_url": "http://mbod.odl.local:8063/c/offeror/see/",
                    "name": "Sloan Executive Education",
                    "professional": true,
                    "offerings": [
                        "Courses",
                        "Programs",
                        "Enterprise or Organizational accounts",
                        "Custom Engagements"
                    ],
                    "audience": [
                        "Global Business Professionals (all levels)",
                        "Entrepreneurs and intrapreneurs",
                        "Managers and executives",
                        "Current and future business leaders"
                    ],
                    "formats": [
                        "Online",
                        "In-Person",
                        "Blended"
                    ],
                    "fee": [
                        "Paid",
                        "Free Webinars and Blogs"
                    ],
                    "certifications": [
                        "Executive Certificates",
                        "Professional Certificates"
                    ],
                    "content_types": [
                        "Professional"
                    ],
                    "more_information": [
                        "https://executive.mit.edu/"
                    ]
                }
            },
```

### How can this be tested?
- Run `docker compose up`
- Run either `.manage.py loadata offered_by` or `./manage.py update_offered_by`, either should work.
- Try out the API endpoints mentioned above.
